src/engine/Engine.cpp

```diff
@@ -1811,6 +1811,15 @@ bool Engine::shouldExitDueToTimeout( unsigned timeout ) const
     return _statistics.getTotalTime() / MILLISECONDS_TO_SECONDS > timeout;
 }
 
+void Engine::reset()
+{
+    Statistics *statistics = new Statistics();
```

Comment:
This looks like a memory leak. Maybe just do "Statistics statistics;", and move this to be inside resetStatistics().